src/main/java16/org/eclipse/yasson/internal/ClassMultiReleaseExtension.java

```diff
@@ -50,7 +51,7 @@ static JsonbCreator findCreator(Class<?> clazz,
                                     AnnotationIntrospector introspector) {
         if (clazz.isRecord()) {
             if (declaredConstructors.length == 1) {
-                return introspector.createJsonbCreator(declaredConstructors[0], null, clazz);
+                return introspector.createJsonbCreator(declaredConstructors[0], null, clazz, pns -> pns);
```

Comment:
I might say, add PropertyNamingStrategy parameter. This class is in internal package so it should be fine to change it. That should fix it even for constructors without creator annotation